### DISCUSSION TITLE
KHR_materials_variants extension with sceneform

### DISCUSSION THREAD

**Discussion by Efor18** [sceneview/sceneform-android#97] [Q&A]
Hi, Is this glTF [KHR_materials_variants](https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_materials_variants) supported by scenform? How can I switch between different textures contained inside the model?

[Here](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/MaterialsVariantsShoe) is an glTF model sample that contains different variants using this material extension.

Can someone help me or point me in the right direction?

Thanks in advance

**Comment by ThomasGorisse** [comment]
Hi,

You can ask if the KHR_materials_variants is currently supported by Filament directly on their [GitHub](https://github.com/google/filament/).
If it's the case, it will be handled by Sceneform.

For changing the texture inside Sceneform, have a look to the [image-texture sample](https://github.com/ThomasGorisse/sceneform-android-sdk/tree/master/samples/image-texture)

**Reply by Efor18** [reply]
Hi @ThomasGorisse,

Thanks for your reply, it looks like is not yet supported by Filament, there is a request for it [here](https://github.com/google/filament/issues/4098)

Do you know if there is a way of changing the texture of just a specific part of the model, not for the whole model? I.e: In a model of a car, change just the texture of the wheels.

**Reply by ThomasGorisse** [reply]
If your wheels are defined in a separate entity inside your glTF, you can change its texture by calling this:
```
model.setRenderable(this.model)
        .getMaterial("wheels")
        .setTexture("baseColorMap", texture);
```
